File: source/python/119. Pascal's Triangle II.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        if rowIndex == 0:
            return [1]

        row = [1, 1]
        for i in range(2, rowIndex + 1):
            row = [1] + [row[j] + row[j + 1] for j in range(i - 1)] + [1]

        return row


def test():
    s = Solution()

    assert s.getRow(0) == [1]
    assert s.getRow(1) == [1, 1]
    assert s.getRow(3) == [1, 3, 3, 1]

    print("All tests passed!")


if __name__ == "__main__":
    test()

File: source/cpp/119. Pascal's Triangle II.cpp
#include <vector>


class Solution {
public:
    std::vector<int> getRow(const int& rowIndex) {
        std::vector<int> result(rowIndex + 1, 1);
        for (auto i = 1; i <= rowIndex; ++i) {
            for (auto j = i - 1; j > 0; --j) {
                result[j] += result[j - 1];
            }
        }
        return result;
    }
};
